x-pack/plugins/lens/public/lens_inspector_service.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  Adapters,
+  Start as InspectorStartContract,
+} from '../../../../src/plugins/inspector/public';
+
+import { createDefaultInspectorAdapters } from '../../../../src/plugins/expressions/public';
+
+export const getLensInspectorService = (inspector: InspectorStartContract) =>
+  new (class InspectorService {
+    public readonly adapters: Adapters = createDefaultInspectorAdapters();
+
+    public inspect = () => {
+      inspector.open(this.adapters);
+    };
+  })();
```

Comment:
We tend to avoid this type of classes construct in Lens.
Maybe something just return an object can be sufficient in this case?
```suggestion
export const getLensInspectorService = (inspector: InspectorStartContract) => {
  const adapters = createDefaultInspectorAdapters();
  return {
    adapters,
    inspect: () => inspector.open(adapters),
  };
};
```